src/Umbraco.Infrastructure/Sync/LastSyncedDatabaseManager.cs

```diff
@@ -0,0 +1,71 @@
+using Microsoft.Extensions.Options;
+using Umbraco.Cms.Core.Configuration.Models;
+using Umbraco.Cms.Core.Hosting;
+using Umbraco.Extensions;
+
+namespace Umbraco.Cms.Infrastructure.Sync;
+
+internal sealed class LastSyncedDatabaseManager : ILastSyncedManager
+{
+    private readonly IHostingEnvironment _hostingEnvironment;
+    private readonly ILastSyncedDatabaseRepository _lastSyncedDatabaseRepository;
+    private HostingSettings _hostingSettings;
+
+    private int? _lastSyncedId;
+
+    public LastSyncedDatabaseManager(
+        IHostingEnvironment hostingEnvironment,
+        IOptionsMonitor<HostingSettings> hostingOptionsMonitor,
+        ILastSyncedDatabaseRepository lastSyncedDatabaseRepository)
+    {
+        _hostingEnvironment = hostingEnvironment;
+        _lastSyncedDatabaseRepository = lastSyncedDatabaseRepository;
+        _hostingSettings = hostingOptionsMonitor.CurrentValue;
+        hostingOptionsMonitor.OnChange(hostingSettings =>
+        {
+            // We sitename changes doing runtime. we save the value to the new name.
+            var lastSyncedId = _lastSyncedId ?? _lastSyncedDatabaseRepository.GetValue(ServerKey);
+            _hostingSettings = hostingSettings;
+
+            _lastSyncedDatabaseRepository.SetValue(ServerKey, lastSyncedId);
+        });
+    }
+
+    private string ServerKey
```

Comment:
Isn't this the same key as used in the `umbracoServer` table and can't this be stored in a new column, removing the need for adding a new table?

Reply:
Machine name is part of the key, which is a part of the problem, when Azure shift worker you get a new machine name.

Reply:
Ah right, so isn't this a similar problem that resulted in adding a `MainDomKeyDiscriminator` in PR https://github.com/umbraco/Umbraco-CMS/pull/12037? Although for the `MainDomId` the machine name and application ID are added (which is what I believe is used in the `umbracoServer` table), but if this discriminator value would serve the same purpose, we might want to use this value instead and avoid requiring users to set multiple values to get things working correctly?

Reply:
Alternatively, why not store the sync state in nucache itself. Then it doesn't matter if files on disc get out of sync with it's state in the sql database. It's already transactional, so the last step of a nucache write could write the lastsyncid.

Reply:
It's used for multiple things, though ideally it would be split.